pydantic/_internal/_constructor_signature_generators.py

```diff
@@ -10,6 +10,13 @@
     from ..fields import FieldInfo
 
 
+def field_to_param_name(field_name: str, field: FieldInfo) -> str:
+    """Convert field to parameter name.
+    Returns the name to use for the parameter, giving priority to the alias of the field.
+    """
```

Comment:
I like that you've consolidated shared logic, but let's come up with a more descriptive name for this function, specifically relating to the fact that we're selecting either the alias or field name for the signature's param name.

Reply:
After considering your previous comments, I don't think that I should duplicate the logic for extracting the alias since it is implemented in the [process_param_defaults()](https://github.com/pydantic/pydantic/blob/3d2ebef8f76625d6f82e6bf417682fd08ac66296/pydantic/_internal/_dataclasses.py#L195) function. Perhaps I should use that function or somehow extract the shared logic between that function and this one. Check out my [comment](https://github.com/pydantic/pydantic/pull/8387#discussion_r1430676077) on your other comments.